src/rf/apps/home/urls.py

```diff
@@ -3,14 +3,30 @@
 from __future__ import unicode_literals
 from __future__ import division
 
-from django.conf.urls import patterns, url
+from django.conf.urls import patterns, url, include
 
-from apps.home.views import home_page
+from rest_framework.routers import DefaultRouter
 
+from apps.home.views import (home_page,
+                             LayerListView,
+                             UserLayerViewSet,
+                             FavoriteListView,
+                             FavoriteCreateDestroyView)
+
+username_regex = r'[\w.@+-]+'
+slug_regex = r'[-_\w]+'
+
+router = DefaultRouter()
+router.register(r'user/(?P<username>' + username_regex + r')/layers',
+                UserLayerViewSet, base_name='user_layers')
 
 urlpatterns = patterns(
-    '',
+    url(r'^', include(router.urls)),
+    url(r'user/(?P<username>' + username_regex + r')/favorites/',
+        FavoriteListView.as_view()),
+    url(r'user/(?P<username>' + username_regex + r')/layers/(?P<slug>' +
+        slug_regex + r')/favorite/',
+        FavoriteCreateDestroyView.as_view()),
+    url(r'layers/', LayerListView.as_view()),
```

Comment:
What's the rationale behind specifying some viewsets via the router but not others?